### PR TITLE
feat(contact groups): add interfaces to find contact groups use case

### PR DESCRIPTION
**Pull request by jeremyjaouen**
## Description

This PR intends to add interfaces to find contact groups use case

**Fixes** # MON-12599

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 20.10.x
- [ ] 21.04.x
- [ ] 21.10.x
- [x] 22.04.x (master)

## Checklist

- [x] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: src/Core/Contact/Application/UseCase/FindContactGroups/FindContactGroups.php [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    }
+
+    /**
+     * @param FindContactGroupsPresenterInterface $presenter
+     */
+    public function __invoke(FindContactGroupsPresenterInterface $presenter): void
+    {
+        try {
+            if ($this->user->isAdmin()) {
+                $contactGroups = $this->repository->findAll();
```
Comment (adr-mo) [inline src/Core/Contact/Application/UseCase/FindContactGroups/FindContactGroups.php:54]: Simple remark as we are currently in the ContactGroup context, is it necessary to define `contactGroups` ?
Could it be only `groups` ?

**Review by adr-mo — APPROVED** [review]

File: src/Core/Contact/Application/UseCase/FindContactGroups/FindContactGroups.php (hunk 1 above)
Comment (jeremyjaouen) [reply src/Core/Contact/Application/UseCase/FindContactGroups/FindContactGroups.php:54]: I think it's more readable to have contactGroups, you don't have to remember in what context you are (even if it's indicate in the class name few lines above)

Comment (adr-mo) [reply src/Core/Contact/Application/UseCase/FindContactGroups/FindContactGroups.php:54]: Seen with @jeremyjaouen let's keep it that way

File: src/Core/Contact/Application/Repository/ReadContactGroupRepositoryInterface.php [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+use Core\Contact\Domain\Model\ContactGroup;
+
+interface ReadContactGroupRepositoryInterface
+{
+    /**
+     * Get all contact groups
+     *
+     * @return array<ContactGroup>
+     */
```
Comment (jeremyjaouen) [inline src/Core/Contact/Application/Repository/ReadContactGroupRepositoryInterface.php:34]: throws Throwable

File: src/Core/Contact/Application/UseCase/FindContactGroups/FindContactGroups.php [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+                        new ForbiddenResponse('You are not allowed to access contact groups')
+                    );
+                    return;
+                }
+                $contactGroups = $this->repository->findAllByUserId($this->user->getId());
+            }
+        } catch (\Throwable $ex) {
+            $this->error('An error occured in data storage while getting contact groups', [
+                'trace' => $ex->getTraceAsString()
+            ]);
```
Comment (jeremyjaouen) [inline src/Core/Contact/Application/UseCase/FindContactGroups/FindContactGroups.php:70]: no need to add trace

File: src/Core/Contact/Application/UseCase/FindContactGroups/FindContactGroups.php [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+            $presenter->setResponseStatus(new ErrorResponse(
+                'Impossible to get contact groups from data storage'
+            ));
+            return;
```
Comment (jeremyjaouen) [inline src/Core/Contact/Application/UseCase/FindContactGroups/FindContactGroups.php:74]:
```suggestion

            return;
```

File: src/Core/Contact/Application/UseCase/FindContactGroups/FindContactGroups.php [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+                $contactGroups = $this->repository->findAll();
+            } else {
+                if (! $this->user->hasTopologyRole(Contact::ROLE_CONFIGURATION_USERS_CONTACT_GROUPS_READ)) {
+                    $this->error('User doesn\'t have sufficient right to see contact groups', [
+                        'user_id' => $this->user->getId(),
+                    ]);
+                    $presenter->setResponseStatus(
+                        new ForbiddenResponse('You are not allowed to access contact groups')
+                    );
+                    return;
```
Comment (jeremyjaouen) [inline src/Core/Contact/Application/UseCase/FindContactGroups/FindContactGroups.php:63]:
```suggestion

                    return;
```

File: src/Core/Contact/Domain/Model/ContactGroup.php [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+{
+    /**
+     * @param integer $id
+     * @param string $name
+     */
+    public function __construct(
+        private int $id,
+        private string $name
+    ) {
+    }
```
Comment (jeremyjaouen) [inline src/Core/Contact/Domain/Model/ContactGroup.php:36]: Assert not empty

**Review by adr-mo — APPROVED** [review]

**Review by a-launois — APPROVED** [review]

**Pull request merged**